frontend-web/src/app/components/sections/Map.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+import { JSX, useEffect, useRef, useState } from 'react';
+
+import Sidebar from '@/app/components/ui/Sidebar';
+import 'leaflet/dist/leaflet.css';
+import { fakeSportVenues } from '@/data/FakeSportVenues';
+
+const Map = (): JSX.Element => {
+  const mapRef = useRef<HTMLDivElement>(null);
+  const [sidebarContent, setSidebarContent] = useState<string | null>(null);
+  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(false);
+
+  const closeSidebar = () => {
+    setIsSidebarOpen(false);
+    setSidebarContent(null);
+  };
+
+  useEffect(() => {
+    if (typeof window === 'undefined' || !mapRef.current) {
+      return;
+    }
+
+    import('leaflet').then(L => {
+      if (mapRef.current!.childElementCount > 0) {
+        return;
+      }
+
+      const customIcon = L.icon({
+        iconUrl: '/images/marqueur.png',
+        iconSize: [30, 30],
+        iconAnchor: [15, 30],
+        popupAnchor: [0, -27],
+      });
+
+      const map = L.map(mapRef.current!).setView([48.8584, 2.2945], 13);
+
+      map.on('click', () => {
+        closeSidebar();
```

Comment:
est-ce que ca distingue bien le clic sur la map, du clic sur les marqueurs et autres boutons ?

Je pense que l'action du clic ne servira pas qu'a fermer la sidebar (peut-etre que plus tard on rajoutera un marqueur perso au double-clic). Je te propose de renommer la fonction en handleSidebar mais de verifier aussi si la sidebar est ouverte.

Donc la premiere action sera de fermer la sidebar si elle est ouverte.
Mais si la sidebar est deja fermée, on devrait pouvoir faire d'autres action :

```suggestion
        handleClick();
```

Reply:
je n'ai pas bien compris se que vous voulez, actuellement si je click sur la map la sidebar se ferme complement, mais si je click sur un marqueur alors quelle ouverte rien ne se passe la sidebar reste ouverte et si je click sur un autre marqueur la sidebar reste ouverte et les informations change juste